src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/DocumentWithDescription.java

```diff
@@ -17,16 +18,19 @@ public DocumentWithDescription(
         Document document,
         String description
     ) {
-        this.document = document;
-        this.description = description;
+        requireNonNull(document);
+        requireNonNull(description);
+
+        this.document = Optional.ofNullable(document);
+        this.description = Optional.ofNullable(description);
     }
 
-    public Document getDocument() {
+    public Optional<Document> getDocument() {
         requireNonNull(document);
```

Comment:
is `requireNonNull` needed? Its set to `Optional` instance at the field level so can it ever be null?

Reply:
Deserializers can bypass constructor so yes they can still be `null`